packages/wp-now/src/download.ts

```diff
@@ -0,0 +1,75 @@
+import request from 'request';
+import fs from 'fs-extra';
+import unzipper from 'unzipper';
+import path from 'path';
+import {
+	SQLITE_FILENAME,
+	SQLITE_PATH,
+	SQLITE_URL,
+	WORDPRESS_ZIPS_PATH,
+	WP_DOWNLOAD_URL,
+	WP_NOW_PATH,
+} from './constants';
+
+async function downloadFile(url: string, dest: string): Promise<void> {
+	return new Promise((resolve, reject) => {
+		const file = fs.createWriteStream(dest);
+		request(url)
+			.pipe(file)
+			.on('finish', () => {
+				file.close(() => resolve());
+			})
+			.on('error', (err: Error) => {
+				fs.unlink(dest);
+				reject(err);
+			});
+	});
+}
+
+async function downloadFileAndUnzip({
+	url,
+	zipPath,
+	unzipPath,
+	checkFinalPath,
+	itemName,
+}) {
+	if (!fs.existsSync(checkFinalPath)) {
+		try {
+			fs.ensureDirSync(path.dirname(zipPath));
+			console.log(`Downloading ${itemName}...`);
+			await downloadFile(url, zipPath);
+			// unzip the file
+			console.log(`Unzipping ${itemName}...`);
+			await fs
```

Comment:
I don't want this comment to stand in the way of the PR, but I think if we wanted we can skip the additional delay of waiting for the file to finish downloading and then starting to unzip it.

with `ReadableStream.tee()` I think we can have the data piping into the file as a download object at the same time it's piping into the unzipper, so that the unzip should finish roughly at the same time as the download.

not a big deal, but could save a few seconds potentially on normal runs.

Reply:
though now I see we're deleting the file right after unzipping, in which case we could also directly pipe the download into the unzipper without worrying about the `tee()`

Reply:
Love the suggestion.
I captured the description in this issue https://github.com/WordPress/wordpress-playground/issues/225
We'll do it as a follow-up PR.

Reply:
I've created this PR: https://github.com/WordPress/wordpress-playground/pull/241/files